src/test/kotlin/uk/gov/justice/digital/hmpps/crimeportalgateway/config/SoapHeaderAddressInterceptorTest.kt

```diff
@@ -0,0 +1,90 @@
+package uk.gov.justice.digital.hmpps.crimeportalgateway.config
+
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.extension.ExtendWith
+import org.mockito.Mockito.*
+import org.mockito.junit.jupiter.MockitoExtension
+import org.springframework.ws.context.MessageContext
+import org.springframework.ws.soap.saaj.SaajSoapMessage
+import javax.xml.soap.*
+import org.mockito.Mockito.`when` as mockitoWhen
+
+@ExtendWith(MockitoExtension::class)
+class SoapHeaderAddressInterceptorTest {
```

Comment:
This is just a clutter of mocks I am afraid. The nature of the beast is to set values on a fairly involved object graph. Best way to do this would be with the integration test.